src/main/kotlin/lotto/view/InputIO.kt

```diff
@@ -1,28 +1,69 @@
 package lotto.view
 
+import lotto.domain.Lotto
 import lotto.domain.LottoNumber
+import lotto.domain.LottoTickets
+import lotto.domain.ManualLottoCount
+import lotto.domain.PurchaseAmount
 
 class InputIO {
-    fun inputPurchaseAmount(inputString: String? = readlnOrNull()): Int {
-        return inputString?.toIntOrNull() ?: 0
+    fun inputPurchaseAmount(inputString: String? = readlnOrNull()): PurchaseAmount {
+        return try {
+            PurchaseAmount(inputString?.toIntOrNull() ?: 0)
+        } catch (e: IllegalArgumentException) {
+            println(e.message)
+            this.inputPurchaseAmount()
+        }
+    }
```

Comment:
재귀 함수를 컴파일러가 반복문으로 변환해주는`tailrec`을 한 번 적용해보셔도 좋을 것 같아요🙂
```suggestion
    tailrec fun inputPurchaseAmount(): PurchaseAmount {
        val amount: Int = readlnOrNull()?.toIntOrNull() ?: 0

        return runCatching { PurchaseAmount(amount) }
            .getOrElse {
                println(it.message)
                inputPurchaseAmount()
            }
    }
```

Reply:
제공해주신 예제 소스에서
 A function is marked as tail-recursive but no tail calls are found
Recursive call is not a tail call
워닝 메세지가 나옵니다. tailrec 키워드가 잘 적용 된것일까요?

Reply:
https://github.com/next-step/kotlin-lotto/pull/794#discussion_r1262335910
해당 코멘트 남겨두었습니다~!